### PR TITLE
Hts fix warnings

### PR DESCRIPTION
**Pull request by Jackie1988**

### CODE REVIEW

File: app/uk/gov/hmrc/helptosaveapi/services/CreateAccountBehaviour.scala [hunk 1]
```diff
@@ -27,28 +27,36 @@ import uk.gov.hmrc.helptosaveapi.models.createaccount._
...
+            }
         }
 
-      case (Some(JsString(registrationChannel)), None, Some(nino)) ⇒
-        collate(json, missingDetails, retrievedUserDetails, nino, registrationChannel)
+      case Some(_) ⇒ Left(ApiValidationError("registration channel is not of expected type String"))
 
-      case (_, _, None) ⇒
-        Left(ApiAccessError())
+      case None    ⇒ Left(ApiValidationError("No registration channel was given", ""))
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosaveapi/services/CreateAccountBehaviour.scala]: This should probably be
```
ApiValidationError("No registration channel was given")
```
instead. Supplying two argument means that we make the `message` field empty here

File: build.sbt [hunk 2]
```diff
@@ -128,7 +128,7 @@ lazy val microservice = Project(appName, file("."))
     evictionWarningOptions in update := EvictionWarningOptions.default.withWarnScalaVersionEviction(false),
     routesGenerator := StaticRoutesGenerator
   )
-  .settings(scalacOptions += "-Xcheckinit")
+  .settings(scalacOptions ++= Seq("-Xcheckinit", "-Ypatmat-exhaust-depth", "off"))
```
Comment (AndyHWChung) [inline build.sbt]: do we need this compile flag anymore?

File: test/uk/gov/hmrc/helptosaveapi/services/HelpToSaveApiServiceSpec.scala [hunk 3]
```diff
@@ -212,6 +212,46 @@ class HelpToSaveApiServiceSpec extends TestSupport with MockPagerDuty {
 
       val fakeRequestWithOnlineRequestWithEmail = FakeRequest().withJsonBody(Json.toJson(onlineRequestWithEmail))
 
+      val createAccountRequestRegChannelIncorrectDataType =
```
Comment (AndyHWChung) [inline test/uk/gov/hmrc/helptosaveapi/services/HelpToSaveApiServiceSpec.scala]: this and the `val` below are only used in one test each - it'd be cleaner to stick these `vals` in their respective tests - I think there's already quite a few `vals` and `defs` sitting outside of the tests

File: app/uk/gov/hmrc/helptosaveapi/services/CreateAccountBehaviour.scala (hunk 1 above)
Comment (Jackie1988) [reply app/uk/gov/hmrc/helptosaveapi/services/CreateAccountBehaviour.scala]: ok, I've removed the second argument

File: build.sbt (hunk 2 above)
Comment (Jackie1988) [reply build.sbt]: No I guess not

File: test/uk/gov/hmrc/helptosaveapi/services/HelpToSaveApiServiceSpec.scala (hunk 3 above)
Comment (Jackie1988) [reply test/uk/gov/hmrc/helptosaveapi/services/HelpToSaveApiServiceSpec.scala]: yeah good idea!

**Comment by AndyHWChung** [comment]
LGTM

**Pull request merged**